### PR TITLE
add glnexus module

### PR DESCRIPTION
**Pull request by ramprasadn**
Add GLnexus module and incorporate in the variant calling subworkflow

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/raredisease/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/raredisease _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: subworkflows/local/deepvariant_caller.nf [hunk 1]
```diff
@@ -5,20 +5,40 @@
...
 workflow DEEPVARIANT_CALLER {
     take:
         bam // channel: [ val(meta), path(bam), path(bai) ]
         fasta // path(fasta)
         fai // path(fai)
+        sample // channel: [ sample, sex, phenotype, paternal_id, maternal_id, case_id ]
 
     main:
         DEEPVARIANT ( bam, fasta, fai )
+        def file_list = DEEPVARIANT.out.gvcf.collect{it[1]}.toList()
```
Comment (projectoriented) [inline subworkflows/local/deepvariant_caller.nf:19]: @ramprasadn `def` can be omitted here as this isn't needed when defining a new channel. You could use this syntax when declaring a function though :)

File: subworkflows/local/deepvariant_caller.nf [hunk 2]
```diff
@@ -5,20 +5,40 @@
...
+                it ->
+                    new_sample_meta = it.clone()
+                    new_sample_meta.id = new_sample_meta.case_id
+                    [ [ 'id':new_sample_meta.id ] ] }
+            .set {case_meta}
+
+        //Combine case meta with the list of gvcfs
+        case_meta.mix(file_list)
+            .collect()
+            .set { ch_gvcfs }
```
Comment (projectoriented) [inline subworkflows/local/deepvariant_caller.nf:34]: Maybe this would work? :) @ramprasadn 

ch_gvcfs = case_meta.mix(file_list)

Comment (ramprasadn) [reply subworkflows/local/deepvariant_caller.nf:34]: Turns out collect is necessary. Without it, you don't get the items emitted as a list, instead, you get case_meta emitted first and then file_list, which doesn't work.

Comment (projectoriented) [inline subworkflows/local/deepvariant_caller.nf:34]:
```suggestion
        ch_gvcfs = case_meta.combine(file_list)
```

I know [combine](https://www.nextflow.io/docs/latest/operator.html?highlight=mix#combine) returns the cartesian product of two channels with more than 1 element each but I don't think we'll encounter that. Perhaps, the combine operator can be a placeholder to simplify the code a bit until we have smth better?

Comment (ramprasadn) [reply subworkflows/local/deepvariant_caller.nf]: Good suggestion! I tested it and it works!

**Review by projectoriented — APPROVED** [review]

**Pull request merged**